### PR TITLE
Adds runtimeclassname support in `options` field for deployment and statefulsets

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Run `make test lint` before submitting a PR
- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]

**Comment by jkhelil** [comment]
@PuneetPunamiya can you add it to unit tests

File: docs/TektonConfig.md [hunk 1]
```diff
@@ -602,17 +602,18 @@ The following fields are supported in `deployment`
       * `nodeSelector` - replaces the existing NodeSelector with this, if not empty
       * `tolerations` - replaces the existing tolerations with this, if not empty
       * `topologySpreadConstraints` - replaces the existing TopologySpreadConstraints with this, if not empty
+      * `runtimeClassName` - adds and updates runtimeClassName
```
Comment (jkandasa) [inline docs/TektonConfig.md:605]: update the same in `StatefulSets` documentation too

Comment (PuneetPunamiya) [reply docs/TektonConfig.md:605]: Done, thanks 👍🏻

**Comment by PuneetPunamiya** [comment]
@jkhelil @jkandasa have added unit tests as well, can you  please take a look at it once 🙂

**Pull request merged**